data/lua/wml-tags.lua

```diff
@@ -727,15 +730,17 @@ function wml_actions.event(cfg)
 		wesnoth.deprecated_message("[event]remove=yes", 2, "1.17.0", "Use [remove_event] instead of [event]remove=yes")
 		wml_actions.remove_event(cfg)
 	else
-		wesnoth.add_event_handler(cfg)
+		local delay = cfg.delayed_variable_substitution
+		if delay == nil then delay = true end
+		wesnoth.game_events.add_wml(delay, cfg)
```

Comment:
Given that `cfg.delayed_variable_substitution` is also handled in the C++, is there a reason to not use just `wesnoth.game_events.add_wml(cfg)`?

I'd prefer to have the C++ only take a fixed number of arguments if possible - I think that was discussed in Discord for a different PR (after this code was already in the current PR).

Reply:
There isn't any reason. It was just slightly simpler because that's what the previous API (where everything was `wesnoth.game_events.add`) did.

I disagree with the general approach of only taking a fixed number of arguments, but I don't mind in this specific case; I can change it.